internal/app/machined/pkg/runtime/v1alpha1/bootloader/sdboot/sdboot.go

```diff
@@ -77,16 +77,16 @@ func ProbeWithCallback(disk string, options options.ProbeOptions, callback func(
 
 	// here we need to read the EFI vars to see if we have any defaults
 	// and populate config accordingly
-	// https://www.freedesktop.org/software/systemd/man/systemd-boot.html#LoaderEntryDefault
-	// this should be set on install/upgrades
+	// https://www.freedesktop.org/software/systemd/man/latest/systemd-boot.html#LoaderEntrySelected
+	// this is set by systemd-boot.
 	efiCtx := efivario.NewDefaultContext()
 
 	bootedEntry, err := ReadVariable(efiCtx, LoaderEntrySelectedName)
 	if err != nil {
 		return nil, err
 	}
 
-	config := &Config{}
```

Comment:
this was really dumb mistake when implementing kexec, always returned a non-nil even if probe was unsuccessful